src/main/java/org/project/Controller/MainController.java

```diff
@@ -124,7 +137,18 @@ public void handleSetClassName(ActionEvent event) {
         if (selectedClassBox != null) {
             String newName = classNameInput.getText();
             TextField className = (TextField) selectedClassBox.getChildren().get(0);
+            String currentName = className.getText();
             className.setText(newName);
+
+            int fromIndex = fromComboBox.getItems().indexOf(currentName);
+            int toIndex = toComboBox.getItems().indexOf(currentName);
+
+            if (fromIndex >= 0) {
+                fromComboBox.getItems().set(fromIndex, newName);
+            }
+            if (toIndex >= 0) {
+                toComboBox.getItems().set(toIndex, newName);
+            }
```

Comment:
This does add the classes as they are being created. But one bug that I found is that it won't delete the option from the combo box as a class in the diagram is deleted

Reply:
I have fixed the bug. Now when the user deletes a class it deletes the class name in the combo box "From" and "To" as well.